### PR TITLE
Fin Commands Test and Fix

### PR DESCRIPTION
**Pull request by sean-e-dietrich**
This is a rewrite of the PR https://github.com/docksal/docksal/pull/618

This includes a test for #526

### CODE REVIEW

**Comment by sean-e-dietrich** [comment]
This is an extension of the following PR but with tests https://github.com/docksal/docksal/pull/618

File: bin/fin [hunk 1]
```diff
@@ -1240,11 +1240,16 @@ show_help_list_commands_in_help ()
 	else
 		addons_path="$HOME/$DOCKSAL_COMMANDS_PATH"
 	fi
-
-	local addons_list=$(find "$addons_path" -type f -perm +100 2>/dev/null)
+	
+	local executable="-executable"
+	if [[ is_mac ]]; then
```
Comment (lmakarov) [inline bin/fin]: This can be a one-liner: `is_mac  && executable="-perm +100"`

File: bin/fin [hunk 2]
```diff
@@ -1240,11 +1240,16 @@ show_help_list_commands_in_help ()
...
+		executable="-perm +100"
+	fi
+	
+	local addons_list=$(find "$addons_path" -type f ${executable} 2>/dev/null)
 	if [ ! -z "$addons_list" ]; then
 		local scope
-		for cmd_name in $(find "$addons_path" -type f -perm +100)
+		for cmd_name in $(find "$addons_path" -type f ${executable})
```
Comment (lmakarov) [inline bin/fin]: `for cmd_name in ${addons_list}`?

File: bin/fin [hunk 3]
```diff
@@ -1240,11 +1240,16 @@ show_help_list_commands_in_help ()
...
-
-	local addons_list=$(find "$addons_path" -type f -perm +100 2>/dev/null)
+	
+	local executable="-executable"
+	if [[ is_mac ]]; then
+		executable="-perm +100"
+	fi
+	
+	local addons_list=$(find "$addons_path" -type f ${executable} 2>/dev/null)
 	if [ ! -z "$addons_list" ]; then
```
Comment (lmakarov) [inline bin/fin]: We are using bash style tests instead of sh, so `if [[ "$addons_list" != "" ]]; then` here

File: tests/smoke-test-general.bats [hunk 4]
```diff
@@ -62,6 +62,22 @@ SKIP=1
 	unset output
 }
 
+@test "fin team/test" {
+	[[ $SKIP == 1 ]] && skip
+
+	cmd='#!/bin/bash
```
Comment (lmakarov) [inline tests/smoke-test-general.bats]: Use heredoc:

```
    cat <<EOF > .docksal/commands/team/test
#!/bin/bash
echo "Test Command"
EOF
```

Comment (lmakarov) [reply tests/smoke-test-general.bats]: Looks like this needs to be applied in several other places in this file

File: tests/smoke-test-general.bats [hunk 5]
```diff
@@ -62,6 +62,22 @@ SKIP=1
 	unset output
 }
 
+@test "fin team/test" {
```
Comment (lmakarov) [inline tests/smoke-test-general.bats]: Let's use `namespace/custom-command` to make it more descriptive

**Comment by achekulaev** [comment]
Thanks Sean 👍

**Pull request merged**